zanata-war/src/main/java/org/zanata/action/DashboardAction.java

```diff
@@ -59,7 +60,8 @@
 
 @Name("dashboardAction")
 @Scope(ScopeType.PAGE)
-@Restrict("#{identity.loggedIn}")
+@ZanataSecured
+@CheckLoggedIn
```

Comment:
* Is @ZanataSecured needed to make @CheckLoggedIn work? 

Yes that's how Seam interceptor works. You need to have a class level annotation to bind it to an/some interceptor(s) then the other annotations (CheckLoggedIn, CheckRole, CheckPermission) are just for that interceptor to perform actual work. Our in-house async framework also work the same way. Once migrated to CDI, we will need to migrate the interceptor to CDI interceptor. 

    * Should we put @Interceptors(ZanataSecurityInterceptor.class) on the @interface CheckLoggedIn too? 

No need to do that. 

    * Is there a sanity test to make sure that classes which need @ZanataSecured actually have it? 
No unfortunately. We could use scannotation or some other tools to scan the entire package on classpath, and check to see any of the class that has security annotations. But the cost may be too high considering our classpath. Unless you want to use external tools like ```find zanata-war/src/main/java/ -type f -name "*.java" | xargs grep "@CheckLoggedIn"``` and ```find zanata-war/src/main/java/ -type f -name "*.java" | xargs grep "@ZanataSecured"``` 

----- Original Message -----

> From: "Sean Flanigan" <notifications@github.com>
> To: "zanata/zanata-server" <zanata-server@noreply.github.com>
> Cc: "Patrick Huang" <pahuang@redhat.com>
> Sent: Wednesday, July 29, 2015 6:04:55 PM
> Subject: Re: [zanata-server] Replace Seam Security (#908)

> In zanata-war/src/main/java/org/zanata/action/DashboardAction.java :
> > @@ -59,7 +60,8 @@
> >
> >  @Name("dashboardAction")
> >  @Scope(ScopeType.PAGE)
> > -@Restrict("#{identity.loggedIn}")
> > +@ZanataSecured
> > +@CheckLoggedIn

> * Is @ZanataSecured needed to make @CheckLoggedIn work?
> * Should we put @Interceptors(ZanataSecurityInterceptor.class) on the
> @interface CheckLoggedIn too?
> * Is there a sanity test to make sure that classes which need @ZanataSecured
> actually have it?

> —
> Reply to this email directly or view it on GitHub .